packages/terra-pills/src/PillList.jsx

```diff
@@ -0,0 +1,297 @@
+   * Providing this prop adds the `aria-label` attribute to the Pill List container element. (Required)
+   */
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * If a visible text label is used with the collection of pills, provide a string of the ID for the html element containing the visible text label.
+   * Supplying the 'ariaLabelledBy' prop will override the 'ariaLabel' prop and adds the `aria-labelledby` attribute instead to the Pill List container element.
+   *
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue) For best practices, ensure the visible text in the html element `id` provided to the Pill List
+   * `ariaLabelledby`prop matches the text provided to the `ariaLabel` prop, for consistency in the case of fallback or errors.
+   */
+  ariaLabelledBy: PropTypes.string,
+  /**
+   * If additional visible information text is used, provide a string containing the IDs for html elements that help describe the intent of the pills and the Pill List container.
+   */
+  ariaDescribedBy: PropTypes.string,
+  /**
+   * The content to be shown in the pills container. Should only contain the sub-component pills.
+   */
+  children: PropTypes.node,
+  /**
+   * Determines if the Pill List is rolled up or not.
+   */
+  isCollapsed: PropTypes.bool,
+  /**
+   * Callback function triggered on click/key press of the roll-up pill
+   */
+  rollUpPillOnTrigger: PropTypes.func,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+};
+
+const defaultProps = {
+  ariaLabelledBy: undefined,
+  ariaDescribedBy: undefined,
+  children: undefined,
+  isCollapsed: false,
+  rollUpPillOnTrigger: undefined,
+};
+
+const PillList = (props) => {
+  const {
+    ariaLabel,
+    ariaLabelledBy,
+    ariaDescribedBy,
+    children,
+    isCollapsed,
+    rollUpPillOnTrigger,
+    intl,
+    ...customProps
+  } = props;
+
+  const theme = React.useContext(ThemeContext);
+  const focusableNodes = [];
+  const pillListRef = useRef();
+  const rollUpPillRef = useRef();
+  const focusableNodeIndex = useRef();
+  const focusedNode = useRef(0);
+  const [rollUpCount, setRollUpCount] = useState(0);
+
+  const handleRollUp = () => {
+    rollUpPillRef.current.removeAttribute('style');
+    let pillListHeight = pillListRef.current.clientHeight;
+    const rollUpPillStyles = window.getComputedStyle(rollUpPillRef.current);
+    const pillHeight = rollUpPillRef.current.offsetHeight + Math.round(parseFloat(rollUpPillStyles.marginTop)) + Math.round(parseFloat(rollUpPillStyles.marginBottom));
+
+    if (pillListHeight <= pillHeight) {
+      rollUpPillRef.current.style.display = 'none';
+      return;
+    }
+
+    const childPills = pillListRef.current.querySelectorAll('[data-terra-pill]');
+    let index = childPills.length - 1;
+    let hiddenPillCounter = 0;
+    while (pillListHeight > pillHeight) {
```

Comment:
Well I might be wrong, looks like this will throw an exception rather than loop forever when the index goes below 0.

```suggestion
    while (pillListHeight > pillHeight && index > -1) {
```